src/channel_state.ts

```diff
@@ -507,8 +574,34 @@ export class ChannelState<
       >['messageToImmutable']
     >,
     timestampChanged = false,
+    sortBy: (
+      msg: ReturnType<
+        ChannelState<
+          AttachmentType,
+          ChannelType,
+          CommandType,
+          EventType,
+          MessageType,
+          ReactionType,
+          UserType
+        >['messageToImmutable']
+      >,
+    ) => ImmutableDate = (msg) => msg.created_at,
```

Comment:
while this is a more powerful approach, it can be simplified to a simple string for the field name: `sortBy  = 'created_at'`

Reply:
TS was complaining when I was trying to do that

Reply:
Any help with that is appreciated of course

Reply:
will look into the error when you updated it, worst case we just silence the TS compiler

Reply:
It works fine now. I guess I had other issue which I fixed in a meanwhile

Reply:
@mahboubii should probably confirm this, but this might work:
```typescript
// changed typedef to string
sortBy = 'created_at'
(...)
const attr = message[sortBy]
if(attr instanceof Date) {
  const messageTime = attr.getTime();
}
```